src/daemon/main.cpp

```diff
@@ -219,7 +219,24 @@ int main(int argc, char const * argv[])
       mlog_set_log(command_line::get_arg(vm, daemon_args::arg_log_level).c_str());
     }
 
-    // after logs initialized
+    // after logs initialized create the data directory. Will check if the default data directory is used and if it exists. Then will ensure that migration from the old data directory (.loki) has occurred if it exists.
+    if (command_line::is_arg_defaulted(vm, cryptonote::arg_data_dir) && !fs::exists(data_dir)){
+      auto old_data_dir = tools::get_depreciated_default_data_dir();
+      if (testnet) old_data_dir /= "testnet";
+      else if (devnet) old_data_dir /= "devnet";
+
+      if (fs::is_directory(old_data_dir))
+      {
+        if (!tools::create_directories_if_necessary(data_dir.parent_path()))
+          throw std::runtime_error{"Data-directory migration failed: can't create " + data_dir.parent_path().u8string()};
+        fs::rename(old_data_dir, data_dir);
+        MGINFO_CYAN("Migrated data directory from " << old_data_dir << " to " << data_dir)
+#ifdef WIN32
```

Comment:
`#ifndef`

Reply:
I actually learned that Windows apparently *does* support symlinks since Vista or so, so I removed this.  (We ignore failure on creating the symlink, so even if it doesn't work its no big deal).